app/actions/websocket/channel.test.ts

```diff
@@ -0,0 +1,577 @@
+import {setCurrentTeamId, getCurrentChannelId, getCurrentTeamId, getConfig} from '@queries/servers/system';
+import {getCurrentUser, getTeammateNameDisplay, getUserById} from '@queries/servers/user';
+import EphemeralStore from '@store/ephemeral_store';
+
+import {handleChannelCreatedEvent, handleChannelUnarchiveEvent, handleChannelConvertedEvent, handleChannelUpdatedEvent, handleChannelViewedEvent, handleMultipleChannelsViewedEvent, handleChannelMemberUpdatedEvent, handleChannelDeletedEvent, handleDirectAddedEvent, handleUserAddedToChannelEvent, handleUserRemovedFromChannelEvent} from './channel';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+
+jest.mock('@database/manager');
+jest.mock('@store/ephemeral_store');
+jest.mock('@actions/local/category');
+jest.mock('@actions/remote/channel');
+jest.mock('@actions/remote/role');
+jest.mock('@actions/remote/post');
+jest.mock('@actions/remote/user');
+jest.mock('@actions/local/channel');
+jest.mock('@actions/local/post');
+jest.mock('@queries/servers/channel');
+jest.mock('@queries/servers/system');
+jest.mock('@queries/servers/user');
+jest.mock('@calls/actions');
+jest.mock('@actions/local/user');
+jest.mock('@utils/log');
+jest.mock('@calls/state');
+jest.mock('@calls/actions/calls');
+
+const serverUrl = 'baseHandler.test.com';
+
+describe('WebSocket Channel Actions', () => {
+    let msg: any;
+    let operator: ServerDataOperator;
+
+    const teamId = 'teamid1';
+    const channelId = 'channelid1';
+    const userId = 'userid1';
+
+    beforeEach(async () => {
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+        msg = {
+            data: {
+                team_id: teamId,
+                channel_id: channelId,
+                channel: JSON.stringify({id: channelId, type: General.PRIVATE_CHANNEL, team_id: teamId}),
+            },
+            broadcast: {
+                channel_id: channelId,
+                userId,
+                user_id: userId,
+            },
+        };
+        DatabaseManager.getActiveServerUrl = jest.fn().mockResolvedValue(serverUrl);
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+        jest.restoreAllMocks();
+    });
+
+    describe('handleChannelCreatedEvent', () => {
+        it('should handle channel created event', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: [{}], memberships: [{}]});
+            (prepareMyChannelsForTeam as jest.Mock).mockResolvedValue([[]]);
+            (addChannelToDefaultCategory as jest.Mock).mockResolvedValue({models: []});
+
+            await handleChannelCreatedEvent(serverUrl, msg);
+
+            expect(getChannelById).toHaveBeenCalled();
```

Comment:
This is more a implementation detail that something to really test.

Reply:
What else would we check for? The function returns nothing.

Reply:
handleChannelCreatedEvent should add the channel in the message to the database, so that should be what we test for.

Testing for that is tricky too, of course. But should be possible to just call getChannelById ourselves on the test. But maybe the heavy mocking may make things not go 100% through. 0/5

Either way, for this particular comment, we can directly skip it. In other places, where we are saying "getChannelById should not have been called", we may want to check other options.

Reply:
Right but then we're not testing what this function is doing we're also testing what other functions are doing. I suppose it comes down to if we want unit or integration tests - which we can discuss in this chain https://github.com/mattermost/mattermost-mobile/pull/8417#discussion_r1888262021